src/domain/wallets/withdrawal-fee-calculator.ts

```diff
@@ -1,13 +1,61 @@
 import { toSats } from "@domain/bitcoin"
 
-export const WithdrawalFeeCalculator = (): WithdrawalFeeCalculator => {
-  const onChainWithdrawalFee = ({
+import { WithdrawalFeePriceMethod } from "."
+
+export const WithdrawalFeeCalculator = ({
+  thresholdImbalance,
+  feeRatio,
+}: OnchainWithdrawalConfig): WithdrawalFeeCalculator => {
+  const onChainWithdrawalFlatFee = async ({
+    minBankFee,
+    minerFee,
+  }: OnChainWithdrawalFlatFeeArgs): Promise<WithdrawalFeeCalculatorRes> => ({
+    totalFee: toSats(minerFee + minBankFee),
+    bankFee: minBankFee,
+  })
+
+  const onChainWithdrawalProportionalOnImbalanceFee = async ({
+    minerFee,
+    minBankFee,
+    imbalanceCalculatorFn,
```

Comment:
I don't see the need to add this fn here. IMO it would be better to simply call the `WithdrawalFeeCalculator` with the already computed imbalance. That would make it more isolated and only depend on values.

In some cases dependency injection is unavoidable but if we can avoid it and just work with values we should. OTOH if you do inject it it should be injected via the `OnChainWithdrawalConfig` constructor args as I don't see this function ever changing at runtime.

Reply:
Also this introduces a circular dependency on the module level from `ledger -> wallet -> ledger`. We need to keep the dependency graph uni-directional `ledger -> wallet`